apps/wallet-mobile/src/features/Receive/common/ShareQRCodeCard/ShareQRCodeCard.tsx

```diff
@@ -0,0 +1,197 @@
+import {useTheme} from '@yoroi/theme'
+import React, {RefObject, useEffect, useRef, useState} from 'react'
+import {StyleSheet, TouchableOpacity, useWindowDimensions, View} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import QRCode from 'react-native-qrcode-svg'
+import Animated, {FadeInDown, FadeOutDown, Layout} from 'react-native-reanimated'
+import Share from 'react-native-share'
+import ViewShot, {captureRef} from 'react-native-view-shot'
+
+import {Text} from '../../../../components'
+import {CaptureShareQRCodeCard} from '../CaptureShareQRCodeCard/CaptureShareQRCodeCard'
+import {mocks} from '../mocks'
+import {useStrings} from '../useStrings'
+
+type ShareProps = {
+  address?: string
+  title?: string
+  addressDetails?: AddressDetailsProps
+  isCopying?: boolean
+  onLongPress?: () => void
+}
+
+type AddressDetailsProps = {
+  address: string
+  stakingHash?: string
+  spendingHash?: string
+  title?: string
+}
+
+export function ShareQRCodeCard({address, title, isCopying, onLongPress}: ShareProps) {
+  const [isSharing, setIsSharing] = useState<boolean>(false)
+  const strings = useStrings()
+  const ref: RefObject<ViewShot> = useRef(null)
+
+  const {styles, colors} = useStyles()
+
+  const shareImage = () => {
+    setIsSharing(true)
+  }
+
+  useEffect(() => {
+    if (isSharing) {
+      const captureAndShare = async () => {
+        try {
+          await new Promise((resolve) => setTimeout(resolve, 50))
```

Comment:
use the delay helper, is it really necessary?

Reply:
Yes, I had to put a delay to have time to screenshot the card with qrcode and share

Reply:
ok, import delay from lodash